core/src/test/scala/org/apache/spark/deploy/StandaloneDynamicAllocationSuite.scala

```diff
@@ -514,10 +518,10 @@ class StandaloneDynamicAllocationSuite
       val scheduler = new CoarseGrainedSchedulerBackend(taskScheduler, rpcEnv)
       try {
         scheduler.start()
-        scheduler.driverEndpoint.ask[Boolean](message)
-        eventually(timeout(10.seconds), interval(100.millis)) {
-          verify(endpointRef).send(RegisterExecutorFailed(any()))
+        intercept[Exception] {
```

Comment:
What about saving the intercepted exception into a `val` (ie ) and checking its content with asserts?

Like:
```
        assert(exception.getCause.getMessage === "Executor is blacklisted: one")
```

Reply:
I very much dislike checking error messages, they are not part of the API contract. But I can try to check for a more specific exception.